### PR TITLE
Add notification links to the sidebar

### PR DESCRIPTION
**Pull request by parrish**

### CODE REVIEW

File: app/talk/lib/sidebar-notifications.cjsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  displayName: 'SidebarNotifications'
+
+  componentDidMount: ->
+    @getUndeliveredCount() if @props.user
+
+  componentWillReceiveProps: ->
+    @getUndeliveredCount() if @props.user
+
+  getInitialState: ->
+    { }
```
Comment (aweiksnar) [inline app/talk/lib/sidebar-notifications.cjsx]: Could this initialize `unreadCount` too? Just for documentation

**Comment by parrish** [comment]
Updated!

File: app/talk/lib/sidebar-notifications.cjsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      @setState {unreadCount}
+
+  label: ->
+    if @state.unreadCount > 0
+      "Notifications (#{ formatNumber @state.unreadCount } unread)"
+    else
+      'Notifications'
+
+  render: ->
+    return <span></span> unless @props.user and @state.unreadCount?
```
Comment (aweiksnar) [inline app/talk/lib/sidebar-notifications.cjsx]: It's also safe to return `null` in these cases if you don't need a specific dom element rendered

**Comment by parrish** [comment]
Updated again!

**Comment by aweiksnar** [comment]
Cool thanks, Looks good! :shipit:

**Comment by parrish** [comment]
Actually, I did need to return an element there apparently:

`Uncaught (in promise) Error: Invariant Violation: SidebarNotifications.render(): A valid ReactComponent must be returned. You may have returned undefined, an array or some other invalid object.(…)`

**Pull request merged**